### PR TITLE
[NSTK -1853] Add validation for values.yaml in helm

### PR DESCRIPTION
**Pull request by david-enli**
Added quotation marks for string objects in values.yaml
Added empty FlashArrays FlashBlades place holders
Added values.json.schema for validation, no that Json schema validation is by default case sensitive, so we might want to communicate with customers.

### CODE REVIEW

File: pure-csi/values.schema.json [hunk 1]
```diff
@@ -0,0 +1,1516 @@
...
+            "clusterrolebinding": {
+                "serviceAccount": {
+                    "name": "pure"
+                }
+            },
+            "csi": {
+                "clusterDriverRegistrar": {
+                    "image": {
+                        "name": "quay.io/k8scsi/csi-cluster-driver-registrar",
+                        "pullPolicy": "IfNotPresent"
```
Comment (sdodsley) [inline pure-csi/values.schema.json:57]: Please make all references to `pullPolicy` to be `Always`. We have had issues in the past with old images being used by customers

Comment (david-enli) [reply pure-csi/values.schema.json:57]: will do, thx

File: pure-csi/values.schema.json [hunk 2]
```diff
@@ -0,0 +1,1516 @@
...
+    "examples": [
+        {
+            "app": {
+                "debug": true
+            },
+            "arrays": {
+                "FlashArrays": [
+                    {
+                        "MgmtEndPoint": "1.2.3.4",
+                        "APIToken": "a526a4c6-18b0-a8c9-1afa-3499293574bb",
```
Comment (dsupure) [inline pure-csi/values.schema.json]: Please replace the APITokens using some dummy examples.

File: pure-csi/values.schema.json [hunk 3]
```diff
@@ -0,0 +1,1516 @@
...
+                "iSCSIAllowedCIDR": "",
+                "iSCSILoginTimeout": 20,
+                "preemptAttachments": "true",
+                "sanType": "ISCSI"
+            },
+            "flashblade": {
+                "snapshotDirectoryEnabled": "false"
+            },
+            "image": {
+                "name": "pc2-dtr.dev.purestorage.com/purestorage/k8s",
```
Comment (dsupure) [inline pure-csi/values.schema.json:106]: Please replace  `"pc2-dtr.dev.purestorage.com/purestorage/k8s"` with `"purestorage/k8s"`

File: pure-csi/values.schema.json [hunk 4]
```diff
@@ -0,0 +1,1516 @@
...
+                "pullPolicy": "Always",
+                "tag": "5.2.0-17e29570-a4b94e748c140859dbf5896525ea9c74"
```
Comment (dsupure) [inline pure-csi/values.schema.json:108]: Please replace "5.2.0-17e29570-a4b94e748c140859dbf5896525ea9c74" with "5.2.0"

File: pure-csi/values.yaml [hunk 5]
```diff
@@ -3,35 +3,35 @@
 # Declare variables to be passed into your templates.
 
 image:
-  name: purestorage/k8s
+  name: "purestorage/k8s"
   tag: 5.2.0
```
Comment (dsupure) [inline pure-csi/values.yaml:7]: The tag shouldn't be a string as well?

File: pure-csi/values.schema.json [hunk 6]
```diff
@@ -0,0 +1,1516 @@
...
+                    "title": "The defaultFSType schema",
+                    "description": "An explanation about the purpose of this instance.",
+                    "default": "",
+                    "examples": [
+                        "xfs"
+                    ]
+                },
+                "defaultMountOpt": {
+                    "$id": "#/properties/flasharray/properties/defaultMountOpt",
+                    "type": "array",
```
Comment (dsupure) [inline pure-csi/values.schema.json:1080]: We previously treat `defaultMountOpt` can be either array or string. I think it is OK to restrict it to array in the future.

Comment (Pure-AdamuKaapan) [reply pure-csi/values.schema.json:1080]: We'll wanna be careful balancing that with backwards-compatibility, but now that it's a much cleaner message I feel like we can make the cut over now.

File: pure-csi/values.schema.json [hunk 7]
```diff
@@ -0,0 +1,1516 @@
...
+                                        "Labels": {
+                                            "topology.purestorage.com/rack": "7b",
+                                            "topology.purestorage.com/env": "dev"
+                                        }
+                                    }
+                                ],
+                                "required": [
+                                    "MgmtEndPoint",
+                                    "APIToken"
+                                ],
```
Comment (carysyd) [inline pure-csi/values.schema.json:453]: I think for FlashBlades, the "NFSEndPoint" is also required.

File: pure-csi/values.schema.json [hunk 8]
```diff
@@ -0,0 +1,1516 @@
...
+                                "oneOf": [
+                                    {
+                                        "type": "object",
+                                        "required": ["NFSEndPoint"]
+                                    },
+                                    {
+                                        "type": "object",
+                                        "required": ["NfsEndPoint"]
+                                    }
+                                ],
```
Comment (carysyd) [inline pure-csi/values.schema.json:463]: For the team to discuss, should we take the opportunity to unify these as well?
Probably just one is fine.

File: pure-csi/values.schema.json [hunk 9]
```diff
@@ -0,0 +1,1516 @@
...
+        "storageclass",
+        "storagetopology"
+    ],
+    "additionalProperties": true,
+    "properties": {
+        "app": {
+            "$id": "#/properties/app",
+            "type": "object",
+            "title": "The app schema",
+            "description": "An explanation about the purpose of this instance.",
```
[the remark this answers is not in the document]
Comment (david-enli) [reply pure-csi/values.schema.json:159]: Yea, I will delete them for now, thanks

**Comment by david-enli** [comment]
functional tests passed, ready for merging

**Pull request merged**